agent/src/test/java/org/openjdk/jmc/agent/test/TestRetrieveEventProbes.java

```diff
@@ -0,0 +1,89 @@
+ * the distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to
+ * endorse or promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ * WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ * WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.openjdk.jmc.agent.test;
+
+import java.lang.management.ManagementFactory;
+
+import javax.management.JMX;
+import javax.management.ObjectName;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openjdk.jmc.agent.jmx.AgentControllerMXBean;
+
+public class TestRetrieveEventProbes {
+
+	private static final String AGENT_OBJECT_NAME = "org.openjdk.jmc.jfr.agent:type=AgentController"; //$NON-NLS-1$
+
+	private static final String XML_TEST_DESCRIPTION = "<jfragent>"
+			+ "<events>"
+			+ "<event id=\"demo.jfr.test1\">"
+			+ "<name>JFR Hello World Event 1 </name>"
+			+ "<description>Defined in the xml file and added by the agent.</description>"
+			+ "<path>demo/jfrhelloworldevent1</path>"
+			+ "<stacktrace>true</stacktrace>"
+			+ "<class>org.openjdk.jmc.agent.test.InstrumentMe</class>"
+			+ "<method>"
+			+ "<name>printHelloWorldJFR1</name>"
+			+ "<descriptor>()V</descriptor>"
+			+ "</method>"
+			+ "<location>WRAP</location>"
+			+ "</event>"
+			+ "</events>"
+			+ "</jfragent>";
+	@Test
+	public void testRetrieveEventProbes() throws Exception {
+		AgentControllerMXBean mbean = JMX.newMXBeanProxy(ManagementFactory.getPlatformMBeanServer(),
+				new ObjectName(AGENT_OBJECT_NAME), AgentControllerMXBean.class, false);
+
+		Assert.assertNotEquals(mbean.retrieveEventProbes(), XML_TEST_DESCRIPTION);
+		mbean.defineEventProbes(XML_TEST_DESCRIPTION);
+		Assert.assertEquals(mbean.retrieveEventProbes(), XML_TEST_DESCRIPTION);
+	}
+
+	@Test
+	public void testRetrieveInvalidConfiguration() throws Exception {
+		AgentControllerMXBean mbean = JMX.newMXBeanProxy(ManagementFactory.getPlatformMBeanServer(),
+				new ObjectName(AGENT_OBJECT_NAME), AgentControllerMXBean.class, false);
+
+		String initialConfiguration = mbean.retrieveEventProbes();
+		String invalidConfiguration = XML_TEST_DESCRIPTION.concat("</jfragent>");
+		mbean.defineEventProbes(invalidConfiguration);
```

Comment:
This is just a matter of taste, but I would have expected the definition to fail exceptionally with information about the invalid configuration, hopefully giving a hint of what went wrong (in the case of e.g. bad XML where the XML went wrong, simply propagating the exception). And then (just as it is done) let subsequent reads of the configuration return the last valid one.

Reply:
Yes, I do agree with you - right now how it is handled is an error is logged when the definition fails, but no exception is thrown.  Since this is more specific to `defineEventProbes` instead of `retrieveEventProbes`, perhaps a separate issue should be created to address this?

Reply:
Ok. I opened up JMC-6890 for this.